planner/core/logical_plan_builder.go

```diff
@@ -2678,6 +2693,68 @@ func (b *PlanBuilder) buildWindowFunction(p LogicalPlan, expr *ast.WindowFuncExp
 	return window, nil
 }
 
+// resolveWindowSpec resolve window specifications for sql like `select ... from t window w1 as (w2), w2 as (partition by a)`.
+// We need to resolve the referenced window to get the definition of current window spec.
+func resolveWindowSpec(spec *ast.WindowSpec, specs map[string]ast.WindowSpec, inStack map[string]bool) error {
+	if inStack[spec.Name.L] {
+		return errors.Trace(ErrWindowCircularityInWindowGraph)
+	}
+	if spec.Ref.L == "" {
+		return nil
+	}
+	ref, ok := specs[spec.Ref.L]
+	if !ok {
+		return ErrWindowNoSuchWindow.GenWithStackByArgs(spec.Ref.O)
+	}
+	inStack[spec.Name.L] = true
+	err := resolveWindowSpec(&ref, specs, inStack)
+	if err != nil {
+		return err
+	}
+	inStack[spec.Name.L] = false
+	return mergeWindowSpec(spec, &ref)
+}
+
+func mergeWindowSpec(spec, ref *ast.WindowSpec) error {
+	if ref.Frame != nil {
+		return ErrWindowNoInherentFrame.GenWithStackByArgs(ref.Name.O)
+	}
+	if ref.OrderBy != nil {
+		if spec.OrderBy != nil {
+			name := spec.Name.O
+			if name == "" {
+				name = "<unnamed window>"
+			}
+			return ErrWindowNoRedefineOrderBy.GenWithStackByArgs(name, ref.Name.O)
+		}
+		spec.OrderBy = ref.OrderBy
+	}
+	if spec.PartitionBy != nil {
+		return errors.Trace(ErrWindowNoChildPartitioning)
```

Comment:
This seems incompatible with MySQL?  According to https://dev.mysql.com/doc/refman/8.0/en/window-functions-named-windows.html, `partition by` clause can be in referenced window while `over` clause can be without it?

Reply:
According to the documentation, MySQL works for queries like:
```
OVER (w ORDER BY country)
... WINDOW w AS (PARTITION BY country)
```
in this case, `partition by` is defined in named window, so `spec.PartitionBy` should be `nil` now, we would raise error for this kind of query?

Reply:
We won't raise error since the check condition is `partitionBy != nil`?

Reply:
Oh my mistake, then how about
```
OVER (w PARTITION BY country)
... WINDOW w AS (ORDER BY country)
```
would MySQL raise error? (I don't have MySQL 8.0 instance ...)

Reply:
Yes, MySQL would raise an error.